presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp

```diff
@@ -431,22 +431,31 @@ void PeriodicTaskManager::addOperatingSystemStatsTask() {
         memset(&usage, 0, sizeof(usage));
         getrusage(RUSAGE_SELF, &usage);
 
+        static int64_t userCpuTimeMicrosOld{0};
```

Comment:
Why not make a member field of PeriodicTaskManager
lastUserCpuTimeMicros_ dittos? Thanks!

Reply:
@xiaoxmeng 
What is the benefit Why do we need them as members?

Reply:
The code is more organized. You can call PeriodicTaskManager's method for this?
```
void updateOSStats() {
  1. refresh;
  2.report delta
   3.update.
}
```

Reply:
@xiaoxmeng 
It is a simple logic - I don't think we need to take out anything (including variables) to the header.
If not for "scheduler_" this method would not even need to be in the class.
Actually PeriodicTaskManager::addOperatingSystemStatsTask() is not even needed.
Instead, where addOperatingSystemStatsTask() is called we should have:
```
scheduler_.addFunction([]() { global::addOsStats(); });
```

But I would land it as is because it is simply not important how exactly it is implemented.

Reply:
@spershin we shall make the implementation logic easy to maintain and extend. Accept this to unblock.